lib/generator.js

```diff
@@ -548,19 +549,54 @@ class Generator {
     } else {
       filename = filenamify(filename, { replacement: '-', maxLength: 255 });
     }
+
     const newFileName = fileName.replace(`\$\$${template}\$\$`, filename);
     const targetFile = path.resolve(this.targetDir, relativeBaseDir, newFileName);
     const relativeTargetFile = path.relative(this.targetDir, targetFile);
-    const shouldOverwriteFile = await this.shouldOverwriteFile(relativeTargetFile);
+    const shouldOverwriteFile = this.shouldOverwriteFile(relativeTargetFile);
     if (!shouldOverwriteFile) return;
     //Ensure the same object are parsed to the renderFile method as before.
     const temp = {};
     const key = template === 'everySchema' || template === 'objectSchema' ? 'schema' : template;
     temp[`${key}Name`] = name;
     temp[key] = component;
-    const content = await this.renderFile(asyncapiDocument, path.resolve(baseDir, fileName), temp);
+    await this.renderAndWriteToFile(asyncapiDocument, path.resolve(baseDir, fileName), targetFile, temp);
+  }
+
+  /**
+   * Render and write to file
+   * 
+   * @private
+   * @param {AsyncAPIDocument} asyncapiDocument AsyncAPI document to pass to the template.
+   * @param {String} templateFilePath Path to the file being rendered.
+   * @param {String} outputpath Path to the file being rendered.
+   * @param {Object} [extraTemplateData] Extra data to pass to the template.
+   */
+  async renderAndWriteToFile(asyncapiDocument, templateFilePath, outputpath, extraTemplateData) {
+    const renderContent =  await this.renderFile(asyncapiDocument, templateFilePath, extraTemplateData);
+    console.log(renderContent);
+    if (this.templateConfig && this.templateConfig.type === 'react') {
+      let filePath = outputpath;
+      let permission = 0o666;
```

Comment:
Should it be `permissions` instead of `permission`?

Reply:
Should be for an individual file not multiple, so 1 permission right? :thinking:

Reply:
never mind you are right.